lxd/storage/drivers/driver_btrfs.go

```diff
@@ -449,6 +449,15 @@ func (d *btrfs) MigrationTypes(contentType ContentType, refresh bool, copySnapsh
 		}
 	}
 
+	if refresh && !copySnapshots {
```

Comment:
Shouldnt this be in this PR https://github.com/lxc/lxd/pull/11368?

Reply:
Originally, I wanted to have 2 separate PRs (one for rsync, the other for various zfs block mode fixes) but I noticed they needed to be merged in order for all tests to succeed. I'll be closing the other one in favour of this one.

Reply:
It would be good to have the rsync one split from this PR. It makes it easier to review and makes it possible to make progress more quickly. I'm not really clear why you can't base this PR off of the separate one for the rsync changes, and then once its merged you can rebase.